Sources/HummingbirdFluent/Fluent.swift

```diff
@@ -14,85 +14,57 @@
 
 import FluentKit
 import Hummingbird
+import ServiceLifecycle
+
+@MainActor
+public struct MainActorBox<Value>: Sendable {
+    public let value: Value
+}
+
+extension DatabaseID: @unchecked Sendable {}
 
 /// Manage fluent databases and migrations
 ///
 /// You can either create this separate from `HBApplication` or add it to your application
 /// using `HBApplication.addFluent`.
-public struct HBFluent {
-    /// Fluent history management
-    public class History {
-        /// Is history recording enabled
-        public private(set) var enabled: Bool
-        // History of queries to Fluent
-        public private(set) var history: QueryHistory?
-
-        init() {
-            self.enabled = false
-            self.history = nil
-        }
-
-        /// Start recording history
-        public func start() {
-            self.enabled = true
-            self.history = .init()
-        }
-
-        /// Stop recording history
-        public func stop() {
-            self.enabled = false
-        }
-
-        /// Clear history
-        public func clear() {
-            self.history = .init()
-        }
-    }
-
-    /// Databases attached
-    public let databases: Databases
-    /// List of migrations
-    public let migrations: Migrations
-    /// Event loop group used by migrator
+public struct HBFluent: Sendable, Service {
+    /// Event loop group
     public let eventLoopGroup: EventLoopGroup
     /// Logger
     public let logger: Logger
-    /// Fluent history setup
-    public let history: History
+    /// List of migrations. Only accessible from the main actor
+    @MainActor
+    public var migrations: Migrations { self._migrations.value }
```

Comment:
I'm currently using `@MainActor` here. @Joannis what are your thoughts on setting up a Hummingbird global actor in Hummingbird and using that instead.

Reply:
Do you foresee other places where you'd want/need this? What's the "domain" of the actor? What type of information do you scope it to?

Reply:
At this point I time probably not so we can keep as `@MainActor`. The migration stuff should only be setup during initialisation.